api/src/main/java/net/kyori/adventure/audience/ForwardingAudience.java

```diff
@@ -74,6 +79,30 @@
     return defaultValue.get(); // unsupported
   }
 
+  @Override
+  default @NonNull Audience filterAudience(final @NonNull Predicate<? super Audience> filter) {
+    @Nullable List<Audience> audiences = null;
+    for(final Audience audience : this.audiences()) {
+      if(filter.test(audience)) {
+        final Audience filtered = audience.filterAudience(filter);
+        if(filtered != Audience.empty()) {
```

Comment:
why call the predicate on an empty audience at all? can this check be moved to before `filter.test(audience)`?